### ISSUE TITLE
RollingFileAppender loss log in multi-process situation

### ISSUE THREAD

**Issue by zzqcraft** [log4cplus/log4cplus#192]
Hi there,

My application will run on different processes at the same time, but we only allow to have single log file. I tried `RollingFileAppender` and launches 2 `fileappender_test` instances simultaneously. I notices that there are some empty lines in the log file where the missing log records are expected to be. For example below, I would expect `479 [380] DEBUG test.subtest <loop> - Entering loop #12768` to be written in the empty line.

```
479 [380] DEBUG test.subtest <loop> - Entering loop #12767
49 [9320] DEBUG test.subtest <loop> - Entering loop #1051

479 [380] DEBUG test.subtest <loop> - Entering loop #12769
49 [9320] DEBUG test.subtest <loop> - Entering loop #1052
```

Is it expected behavior? If yes, is there any config setting I can try to eliminate the loss? My test environment is Windows 10 and Visual Studio 2013.

Thanks,

**Comment by joseAndresGomezTovar** [comment]
The issue here is : there are two proccess accesing At the same  time to
the xxx.log file

When the appender try to rename/create a new one  "fail" and reuse the
same  file adding lines on  top

Try adding socket appender and check  the server socket logs , should be ok

**Comment by wilx** [comment]
@zzqcraft There are two problems with logging from multiple processes into a single file.

First is generic, you have to use locking mechanism to synchronize the writing between the individual processes. This is handled by setting `UseLockFile` property to `true` in you configuration file on your `RollingFileAppender`.

Unfortunately, there is also a Windows specific problem of files being locked for renaming and deletion when they are open. This is hard to work around in log4cplus itself. Because one process can keep a handle open to your log file, it is impossible for the other process to roll the file. There is currently no solution for this. You are basically forced to log into different files on Windows if you want file rolling.

**Comment by zzqcraft** [comment]
Thanks, guys. I will definitely turn on "UseLockFile" property and try it again. 

Talking about file rollover problem. Is that possible to configure the log such that we do not rename old log files but instead creating new file as the new log file? For example, if the log has rolled over 2 times and there are three log files in the system: file.log file1.log file2.log, and "file2.log" is the newest log. By doing this, we do not need to worry about the renaming, and keep the old log as it is.

What do you guys think?

**Comment by joseAndresGomezTovar** [comment]
This feature is not available

**Comment by zzqcraft** [comment]
Thanks a lot

**Comment by jameysiddiqui** [comment]
Is this issue resolved in 2.0.1 release or still the issue exists, two process accessing the same log file.
Please let me know.
Thanks.

**Comment by wilx** [comment]
[**@jameysiddiqui**](https://github.com/jameysiddiqui) commented on [Jun 4, 2018, 6:53 PM GMT+2](https://github.com/log4cplus/log4cplus/issues/192#issuecomment-394424291 "2018-06-04T16:53:07Z - Replied by Github Reply Comments"):
> Is this issue resolved in 2.0.1 release or still the issue exists, two process accessing the same log file.
> Please let me know.
> Thanks.

Did you use the file locking through (if you are using configuration file) [`UseLockFile`](https://github.com/log4cplus/log4cplus/blob/master/include/log4cplus/fileappender.h#L73) property?
